modules/mono/csharp_script.cpp

```diff
@@ -551,8 +551,11 @@ bool CSharpLanguage::handles_global_class_type(const String &p_type) const {
 
 String CSharpLanguage::get_global_class_name(const String &p_path, String *r_base_type, String *r_icon_path) const {
 	Ref<CSharpScript> scr = ResourceLoader::load(p_path, get_type());
-	if (!scr.is_valid() || !scr->valid || !scr->global_class) {
-		// Invalid script or the script is not a global class.
+	// The script is allowed to not be a global class
+	// See modules/gdscript/gdscript.cpp:2499
```

Comment:
This is correct, but the String returned must be empty if it's not a global class. As I understand it, this method should always return the correct `r_base_type` so the early returns are incorrect.

Consider changing the comment to something like:

```suggestion
	// This method must always return the r_base_type for every script, even if the script is
	// not a global class. But if the script is not a global class it must return an empty string
	// since it should not have a name.
```

Also, you will need to make two more changes to this method:

- Remove the next [if that checks if the name is empty and early returns](https://github.com/godotengine/godot/blob/541674d10692cf608f1245d7eede459bdc96b74f/modules/mono/csharp_script.cpp#L560-L562).
- Add a check [before returning the name](https://github.com/godotengine/godot/blob/541674d10692cf608f1245d7eede459bdc96b74f/modules/mono/csharp_script.cpp#L587) like:

```c++
if (!scr->global_class) {
	return String();
}
```